feat: add AWS Bedrock GovCloud model support

colesmcintosh opened the pull request:
## Title

Add AWS Bedrock GovCloud model support

## Relevant issues

Fixes LIT-257

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

🆕 New Feature

## Changes

### Summary
Added support for AWS Bedrock models in GovCloud regions (us-gov-east-1 and us-gov-west-1) to enable government customers to use LiteLLM with FedRAMP High and DoD IL-4/5 approved models.

### What changed
1. **Model Configuration**: Added 18 GovCloud-specific model entries to `model_prices_and_context_window.json` (9 models per region)
2. **Filtering Logic**: Updated `is_bedrock_pricing_only_model()` in `litellm/__init__.py` to allow GovCloud models while still filtering out other regional pricing entries
3. **Test Coverage**: Created new test file `tests/llm_translation/test_bedrock_govcloud.py` with comprehensive tests

### Models Added
The following FedRAMP High and DoD IL-4/5 approved models are now available in both GovCloud regions:
- **Claude Models**: Claude 3.5 Sonnet (anthropic.claude-3-5-sonnet-20240620-v1:0), Claude 3 Haiku (anthropic.claude-3-haiku-20240307-v1:0)
- **Llama Models**: Llama 3 8B (meta.llama3-8b-instruct-v1:0), Llama 3 70B (meta.llama3-70b-instruct-v1:0)
- **Titan Models**: Titan Text Lite/Express/Premier, Titan Embed v1/v2

### Test Results
All new tests pass:
```
LITELLM_LOCAL_MODEL_COST_MAP=True python -m pytest tests/llm_translation/test_bedrock_govcloud.py -v
======================= 14 passed, 17 warnings in 0.15s ========================
```

colesmcintosh committed:
feat: add AWS Bedrock GovCloud model support (LIT-257)

- Added 18 GovCloud-specific model entries (9 per region) to model_prices_and_context_window.json
- Updated is_bedrock_pricing_only_model() to allow GovCloud models (us-gov-east-1, us-gov-west-1)
- Added comprehensive test suite for GovCloud model support
- Ensures GovCloud models use appropriate APIs (Converse for Claude/Llama, Invoke for Titan)

Models added:
- Claude 3.5 Sonnet and Claude 3 Haiku (FedRAMP/IL4/5 approved)
- Llama 3 8B and 70B (FedRAMP/IL4/5 approved)
- Amazon Titan Text and Embedding models

krrishdholakia commented on litellm/__init__.py:
```diff
@@ -500,6 +501,17 @@ def is_bedrock_pricing_only_model(key: str) -> bool:
+        if len(parts) >= 2:
+            region = parts[1]
+            # GovCloud regions should NOT be excluded
+            if region in ["us-gov-east-1", "us-gov-west-1"]:
```
why are these not to be excluded? @colesmcintosh 


i don't believe you can directly call `bedrock/us-gov-east-1/amazon.titan-embed-text-v1`

colesmcintosh replied:
@krrishdholakia These GovCloud models should not be excluded because they are real, callable models that follow the existing `bedrock/region/model` pattern already used in LiteLLM. The format `bedrock/us-gov-east-1/model-name` works exactly like other regional models in the codebase (e.g.,`bedrock/ap-northeast-1/anthropic.claude-v1`)

colesmcintosh committed:
Merge remote-tracking branch 'origin/main' into feature/track-bedrock-gov-cloud-models

colesmcintosh committed:
fix: add bedrock_converse GovCloud model mappings for Claude models

Added missing bedrock_converse model entries for AWS GovCloud regions:
- bedrock_converse/us-gov-east-1/anthropic.claude-3-5-sonnet-20240620-v1:0
- bedrock_converse/us-gov-east-1/anthropic.claude-3-haiku-20240307-v1:0
- bedrock_converse/us-gov-west-1/anthropic.claude-3-5-sonnet-20240620-v1:0
- bedrock_converse/us-gov-west-1/anthropic.claude-3-haiku-20240307-v1:0

This fixes test failures where supports_tool_choice() returned True but
the models weren't properly mapped in the configuration files.

colesmcintosh committed:
Merge branch 'BerriAI:main' into feature/track-bedrock-gov-cloud-models

colesmcintosh committed:
fix: correct AWS GovCloud Bedrock model pricing and configurations

- Fix Claude 3.5 Sonnet pricing (3.6e-06 input, 1.8e-05 output)
- Fix Claude 3 Haiku pricing (3e-07 input, 1.5e-06 output)
- Update Claude 3.5 Sonnet max_tokens from 4096 to 8192
- Add bedrock_converse entries for Llama models with correct token limits
- Add Amazon Nova Pro model for both GovCloud regions
- Add supports_pdf_input flag to Claude models

colesmcintosh committed:
fix: handle bedrock_converse prefix in get_non_litellm_routing_model_name

Fixes test failure where bedrock_converse/region/model paths were not properly
stripped to get the base model name, causing supports_function_calling to
return false for regional bedrock_converse models.

colesmcintosh committed:
Merge branch 'BerriAI:main' into feature/track-bedrock-gov-cloud-models

colesmcintosh committed:
revert: reset bedrock/common_utils.py to match main branch

Remove bedrock_converse prefix handling from get_non_litellm_routing_model_name
to align with main branch implementation.

colesmcintosh committed:
revert: reset litellm/__init__.py to match main branch

- Remove public_model_groups variables
- Remove GovCloud exception handling in is_bedrock_pricing_only_model
- Fix comment formatting

colesmcintosh committed:
revert: reset litellm/__init__.py to exact main branch content

Copy exact content from origin/main with no modifications

colesmcintosh committed:
fix: remove bedrock_converse prefixed models from pricing files

- Remove 10 bedrock_converse entries from model_prices_and_context_window.json
- Remove 4 bedrock_converse entries from litellm/model_prices_and_context_window_backup.json
- These were GovCloud-specific entries that are no longer needed

colesmcintosh committed:
fix: correct AWS GovCloud Bedrock model pricing and configurations

- Fix Anthropic Claude 3.5 Sonnet pricing: $3.60/$18.00 per million tokens (was $3.00/$15.00)
- Fix Anthropic Claude 3 Haiku pricing: $0.30/$1.50 per million tokens (was $0.25/$1.25)
- Fix Claude 3.5 Sonnet max_tokens: 8192 (was 4096)
- Fix Llama model max_tokens: 2048 (was 8192) and max_input_tokens: 8000 (was 8192)
- Fix Llama3-8b output pricing: $2.65 per million tokens (was $0.60)
- Add missing Amazon Nova Pro models for both GovCloud regions
- Add supports_pdf_input flag to Llama models

Based on official AWS Bedrock pricing documentation for GovCloud regions

colesmcintosh committed:
test: fix GovCloud bedrock models test to match implementation

Update test_govcloud_model_in_bedrock_models_list to correctly verify that
GovCloud models are excluded from bedrock_models list as they are
pricing-only models following the bedrock/<region>/<model> pattern.

colesmcintosh committed:
Merge remote-tracking branch 'upstream/main' into HEAD

# Conflicts:
#	litellm/model_prices_and_context_window_backup.json
#	model_prices_and_context_window.json

colesmcintosh committed:
Merge branch 'BerriAI:main' into feature/track-bedrock-gov-cloud-models
